DOCS-7580 checklist info

paigehargrave opened the pull request.

Commit:
New prereq file

Commit:
Add general prereqs

Commit:
Info moved

Commit:
Add environment info

Commit:
Add life-cycle and scaling info

Commit:
Moved info

Commit:
Placeholder info

Commit:
Title update

Commit:
Add new topics to nav

Commit:
Reorder topics

Commit:
Topic titles

Commit:
Moving info

Commit:
Moved info in

Commit:
Consolidation

Commit:
Content re-org

Commit:
Minor wording

Commit:
Link to burst info

Commit:
Link to bursting info

Commit:
Move info

Commit:
Outstanding questions added

Commit:
Wording cleanup

sundippatel commented on modules/ROOT/nav.adoc:
```diff
@@ -1,9 +1,11 @@
- ** xref:install-sys-reqs.adoc[Verify System Requirements for Anypoint Runtime Fabric]
- ** xref:install-port-reqs.adoc[Network Port Requirements for Anypoint Runtime Fabric]
- ** xref:runtime-fabric-limitations.adoc[Limitations]
+ ** xref:install-prereqs.adoc[Installation Prerequisites]
```
Given the changes, I'd be easier to review if we could stage this or render the site locally.

From a general sense, I think all of these pages can (or maybe should) be in a single page that lives in a topic called "Install Runtime Fabric"

paigehargrave replied:
Will consolidate these topics.

Commit:
Consolidate prereq info

Commit:
Moved info

Commit:
Consolidate info

Commit:
Consolidate info

Commit:
Consolidate info

Commit:
Prereqs info

Commit:
Moved info

Commit:
Title update

Commit:
Add page alias

Commit:
Re-org

Commit:
Cleanup

Commit:
Consolidating info

Commit:
Clean up

Commit:
Name update

Commit:
Name update

Commit:
Formatting

Commit:
Formatting

Commit:
New file

Commit:
Moved info

Commit:
Moved info

Commit:
Moved info

Commit:
Moved info

sundippatel committed:
WIP - added table of contents structure

WIP. minor edits and included a TOC to help structure this page better.
I think we lost the network and port requirements.
Once we get all the info in this page, and outline it to the TOC, it'll be easier to condense

Commit:
Consolidate network info

Commit:
Org of content

Commit:
Wording

Commit:
Update per feedback

Commit:
Update per feedback

Commit:
Formatting

Commit:
Formatting

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,425 @@
+endif::[]
+:page-aliases: install-port-reqs.adoc
+
+Review the following requirements with your operations and IT teams to ensure all prerequisites are met before installing Anypoint Runtime Fabric. The Runtime Fabric installation process fails if the minimum requirements are not met.
```
```suggestion
Review the following requirements with your operations and IT teams to ensure that all prerequisites are met before installing Anypoint Runtime Fabric; otherwise, your installation will fail.
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,425 @@
+ 
+See xref:architecture.adoc[Runtime Fabric Architecture] for a description of development and production configurations.
+
+= Internal notes
```
I find this a little awkward, and had no idea what "internal notes" meant until I saw the whole topic set.
What this really is is a TOC for this topic set, which I suggest we present instead as a plain text list within the topic set introduction:

"The following list names the categories of requirements that you must meet before you install Anypoint Runtime Fabric. Details follow the list.

* User skills
* Planning
* Infrastructure (VMs, disks, swap memory, and so on)
* OS (RedHat and CentOS versions)
* Network
** TCP and UDP
** Other network requirements
* Mule runtime engine license and access
* Installation script"

kbostrom replied:
I then suggest retitling the subtopics to match the words used for each list item.

The remainder of this review will be copyediting only.

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,425 @@
+* Mule License Requirements
+** Access to the Mule
+* Installation Requirements
+** More of the procedure the installation script will take.
```
Replace with something like the shorter and simpler list above.

Commit:
Remove temporary TOC

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+If needed, contact your MuleSoft representative for assistance.
+ 
+See xref:architecture.adoc[Runtime Fabric Architecture] for a description of development and production configurations.
+
```
There used to be a text section here that was a sort of TOC; it was labeled "internal," but I suggested a reworking. Was it removed or did I somehow delete it?
Part of the reworking included slight rewordings to the level == requirements headings; I'll move ahead and try to reconstruct.

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+ 
+See xref:architecture.adoc[Runtime Fabric Architecture] for a description of development and production configurations.
+
+== General Knowledge
```
```suggestion
== Installer Experience
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+See xref:architecture.adoc[Runtime Fabric Architecture] for a description of development and production configurations.
+
+== General Knowledge
+Knowledge in the following areas is required to perform a Runtime Fabric installation:
```
```suggestion
To successfully install Runtime Fabric, you must be familiar with the following:
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+Knowledge in the following areas is required to perform a Runtime Fabric installation:
+
+* Linux system commands
+* Working in Linux with root access. Installation requires root access (permission to run the `sudo` command) on all servers in your environment
```
GLOBAL formatting requirement for bullet lists:
```suggestion
* Working in Linux as the root user, with permission to run `sudo`
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+
+* Linux system commands
+* Working in Linux with root access. Installation requires root access (permission to run the `sudo` command) on all servers in your environment
+* If running on AWS, working with Terraform
```
```suggestion
* If installing in an Amazon Web Services (AWS) platform, working with Terraform
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+* If running on AWS, working with Terraform
+* Remote connection mechanisms, such as SSL and SSH
+
+== General Considerations
```
```suggestion
== Installation Environment Planning
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+* Remote connection mechanisms, such as SSL and SSH
+
+== General Considerations
+Ensure you plan for:
```
```suggestion
Ensure that you plan for:
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+== General Considerations
+Ensure you plan for:
+
+* Creating separate Runtime Fabrics to separate production and non-production applications
```
```suggestion
* Creating separate Runtime Fabric instances for production and nonproduction applications
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+Ensure you plan for:
+
+* Creating separate Runtime Fabrics to separate production and non-production applications
+* Creating separate Runtime Fabrics based upon environment network scope, such as DMZ versus private network 
```
```suggestion
* Creating separate Runtime Fabric instances for each network scope, such as DMZ versus private network 
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+
+* Creating separate Runtime Fabrics to separate production and non-production applications
+* Creating separate Runtime Fabrics based upon environment network scope, such as DMZ versus private network 
+* The Runtime Fabric default pod and services CIDR block not interfering with the regions network
```
```suggestion
* Ensuring that Runtime Fabric default pod and services CIDR block do not interfere with the regions network
```
the regions network:  do you mean the region's network? the regions' networks? the network of regions?

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+* Creating separate Runtime Fabrics based upon environment network scope, such as DMZ versus private network 
+* The Runtime Fabric default pod and services CIDR block not interfering with the regions network
+
+== Infrastructure Requirements
```
```suggestion
== Infrastructure Preparation
```
Each of these subtitles is an implied "Requirement," so I would either include the term in each title or none, rather than only some.

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+Perform the following actions:
+
+* Disable swap memory.
+* Ensure chrony (system clocks) status is in sync.
```
```suggestion
* Ensure that chrony (system clocks) status is in sync.
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+
+* Disable swap memory.
+* Ensure chrony (system clocks) status is in sync.
+* Provision the required hardware, or have enough quota with your infrastructure provider to create the required hardware. 
```
```suggestion
* Either provision the required hardware or have enough quota with your infrastructure provider to create the required hardware, including:
```
Is "have enough quota" the way our audience would say this?

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+* Ensure chrony (system clocks) status is in sync.
+* Provision the required hardware, or have enough quota with your infrastructure provider to create the required hardware. 
++
+This includes:
```
```suggestion

```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
++
+This includes:
+
+** x86 or x64 architecture. Only x86 and x64 architectures are supported.
```
Required list-item format:
```suggestion
** x86 or x64 architecture
+ + Only x86 and x64 architectures are supported.
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+** Virtual machines
+** Disks with provisioned IOPS
+** Virtual networks
+** Firewall rules / security groups
```
```suggestion
** Firewall rules or security groups
```
...or "and": just not "and/or" per style guide.

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+** Virtual networks
+** Firewall rules / security groups
+** Load balancers
+* If provisioning hardware with AWS or Azure, you must have the right permissions to create the required resources.
```
parallel list items at first level:
```suggestion
* If provisioning hardware with AWS or Azure, permissions to create the required resources
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+** Load balancers
+* If provisioning hardware with AWS or Azure, you must have the right permissions to create the required resources.
++
+[NOTE]
```
This notation seems out of place to me, interrupting a fairly terse list (Reference topic) with a description of provisioning and performance (Concept).
The list item clearly states the requirement; the reason for that requirement is a separate topic...unless you want to add reasons for each list item?

kbostrom replied:
[Following edits are for reference only; suggest moving this info elsewhere so as to not interrupt list flow.]

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
++
+[NOTE]
+Dedicated disks are required to ensure optimum performance. The Runtime Fabric installer formats
+and mounts the disks needed during installation.
```
```suggestion
and mounts the required dedicated disks during installation.
```

kbostrom replied:
Ah! So, I think I might have a solution to this contextual interruption. How about something like this:

"* If provisioning hardware with AWS or Azure, permissions to create the required resources if the default behavior of AWS and Azure provisioning scripts is not sufficient to accommodate your organization's requirements"
...and then go straight to the next list item, "Know how to forward logs...."

If further explanation of creating alternatives to the default behavior is necessary, this might need a separate topic or section, as mentioned earlier.

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+[NOTE]
+Dedicated disks are required to ensure optimum performance. The Runtime Fabric installer formats
+and mounts the disks needed during installation.
++
+The default behavior of the AWS and Azure provisioning scripts creates a set of virtual machines and disks defined by 
+the minimum requirements. It also creates a private network with the required network ports configured. This is optimal 
+when evaluating Runtime Fabric before integrating within your primary network. You should consult with your 
+network administrator to determine if the default behavior is compatible. You might need to modify the 
+provisioning scripts to accommodate your organization's requirements.
```
See suggestion above for reducing the list interruption created by this explanation.

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+when evaluating Runtime Fabric before integrating within your primary network. You should consult with your 
+network administrator to determine if the default behavior is compatible. You might need to modify the 
+provisioning scripts to accommodate your organization's requirements.
+* Gain an understanding of how to forward logs to your organization's logging provider.
```
```suggestion
* Learn how to forward logs to your organization's logging provider, if you do not know this already.
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+* Gain an understanding of how to forward logs to your organization's logging provider.
+* Obtain your organization's SMTP provider information so that you can configure alerts.
+
+* Verify the following items for your network:
```
```suggestion
* Verify the following network requirements:
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+** Outbound access is allowed using websockets (via port 443) to Anypoint Platform.
+** Your organization's network and security policies, including:
+
+*** Internal network IP address range in CIDR notation
```
```suggestion
*** An internal network IP address range, in CIDR notation
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+** Your organization's network and security policies, including:
+
+*** Internal network IP address range in CIDR notation
+*** The subnet range used for Runtime Fabric in CIDR notation
```
```suggestion
*** A subnet range used for Runtime Fabric, in CIDR notation
```

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+*** Internal network IP address range in CIDR notation
+*** The subnet range used for Runtime Fabric in CIDR notation
+*** If virtual machines are allowed, public IP addresses
+*** How to gain shell access to each VM
```
```suggestion
*** Shell access to each VM
```
OR
```suggestion
*** Ability to gain shell access to each VM
```

kbostrom replied:
Whew! Please doublecheck, when done implementing edits, that list items are worded to logically follow their intro texts and that they are parallel to one another at each level of list item.

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+*** The subnet range used for Runtime Fabric in CIDR notation
+*** If virtual machines are allowed, public IP addresses
+*** How to gain shell access to each VM
+** Determine if inbound traffic should be enabled for Anypoint Runtime Fabric. This is based on the function of your organization's Mule applications. To enable inbound traffic, you must:
```
The intro to this list item at this level is "Verify the following list requirements," so this should be worded...:
```suggestion
** Inbound traffic is enabled for Anypoint Runtime Fabric, if necessary. 
+
+ This is based on the function of your organization's Mule applications. To enable inbound traffic, you must:
```

kbostrom replied:
So, again, the list now departs into a subtopic of HOW to meet a requirement, rather than just listing a requirement.
I know that this is an effect of your attempt to meet audience requests to combine previously separate topics, but I think that this effort might require you to omit or move the "How to" instructions (separate Task topic) and the "Why" explanations (Concepts) from this "Requirements" list (Reference).

kbostrom replied:
What this means is that we really should omit "To enable inbound traffic, you must:" and the steps following, so that this subtopic about Infrastructure preparation or requirements ends with inbound traffic being enabled, if necessary. Period.

kbostrom commented on modules/ROOT/pages/install-prereqs.adoc:
```diff
@@ -0,0 +1,404 @@
+*** If virtual machines are allowed, public IP addresses
+*** How to gain shell access to each VM
+** Determine if inbound traffic should be enabled for Anypoint Runtime Fabric. This is based on the function of your organization's Mule applications. To enable inbound traffic, you must:
+*** Have a TLS certificate with the desired Common Name set to route traffic to Runtime Fabric.
+*** Be able to create or configure an external TCP load balancer to manage traffic to applications running in Runtime Fabric.
+*** Be able to configure DNS settings to associate the Common Name on the TLS certificate to the IP address of the external load balancer.
```
Suggest omitting/moving

kbostrom reviewed (approved):
Edit up to Operating System Requirements

Commit:
Formatting

Commit:
Add page alias

Commit:
Link update

Commit:
Link update

Commit:
Link update

Commit:
Link update

Commit:
Link update

Commit:
Link update

Commit:
Link update

Commit:
Link update

Commit:
Link update

Commit:
Link update

Commit:
Link update

Commit:
Link fix

Commit:
Table formatting

Commit:
Table formatting

Commit:
Formatting

Commit:
Formatting

Commit:
Formatting

Commit:
Formatting

Commit:
Formatting

Commit:
Formatting

Commit:
Formatting

Commit:
Retrofit changes from previous PR (217)

Commit:
Formatting

Commit:
Retrofit info from 219

Commit:
Typo

Commit:
Retrofit info from 221

Commit:
Retrofit 206 updates

Commit:
Retrofit 215 changes

Commit:
Retrofit 213 updates

Commit:
Retrofit

Commit:
Retrofit
